packages/react-data-grid-addons/src/data/RowGrouper.js

```diff
@@ -21,7 +21,7 @@ class RowGrouper {
 
   groupRowsByColumn(rows, columnIndex = 0) {
     let nextColumnIndex = columnIndex;
-    let columnName = this.columns[columnIndex];
+    let columnName = this.columns.length > 0 && typeof this.columns[columnIndex] === 'string' ? this.columns[columnIndex] : this.columns[columnIndex].key;
```

Comment:
yeah - if you render the grid before grouping which I want to - columns here are the grouped columns not all grid columns